### PR TITLE
test: removing unnecessary optionals and increasing branch cov - 1

### PR DESCRIPTION
**Pull request by suyashpatil78**
### <samp>🤖 Generated by Copilot at 881e2a6</samp>

Added and updated test cases for add-edit-expense page and fixed compatibility issues with older devices. The test cases cover new scenarios for transaction fields, corporate card expenses, and autofill settings. The compatibility issues were caused by the optional chaining operator in `taxGroupControl.value` and `currencyObjControl.value` expressions in `add-edit-expense.page.ts`.

### <samp>🤖 Generated by Copilot at 881e2a6</samp>

> _We test the edge cases of our code_
> _We don't rely on optional chaining_
> _We mock the data and fix the format_
> _We are the masters of expense tracking_

### <samp>🤖 Generated by Copilot at 881e2a6</samp>

*  Remove optional chaining operator from `taxGroupControl.value` and `currencyObjControl.value` expressions in `AddEditExpensePage` class to avoid runtime errors on older devices ([link](https://github.com/fylein/fyle-mobile-app/pull/2503/files?diff=unified&w=0#diff-492aac4c505a9ef8fb102b40f692554c3d0dc493b7abddc97b6665166efcc83bL589-R597))
*  Add mock data file `expense-field-obj.data` to import statement in `add-edit-expense-1.spec.ts` for testing transaction fields ([link](https://github.com/fylein/fyle-mobile-app/pull/2503/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L15-R15))
*  Add new test case for `removeCorporateCardExpense` method when `etxn$` observable is undefined in `add-edit-expense-1.spec.ts` and expect it to call `goBack` method ([link](https://github.com/fylein/fyle-mobile-app/pull/2503/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6R692-R709))
*  Add new test case for `getActionSheetOptions` method when `project_id` is defined in `add-edit-expense-1.spec.ts` and expect it to return six action sheet options and verify their handlers ([link](https://github.com/fylein/fyle-mobile-app/pull/2503/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6R1041-R1094))
*  Add new test case for `getNewExpenseObservable` method when autofill settings are disabled and recently used currency is undefined in `add-edit-expense-3.spec.ts` and expect it to return a new expense observable with home currency and source ([link](https://github.com/fylein/fyle-mobile-app/pull/2503/files?diff=unified&w=0#diff-8fde0d56f6a9cf999839c0931458cdae9f882355edcd31b9b700f9fc9e538647R1235-R1262))
*  Change formatting of `TestBed.inject`, `corporateCreditCardExpenseService`, `transactionService`, `popoverController`, `trackingService`, `orgSettingsService`, and `launchDarklyService` statements in `add-edit-expense-1.spec.ts` to remove trailing commas and align closing parentheses for readability and consistency ([link](https://github.com/fylein/fyle-mobile-app/pull/2503/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L147-R151), [link](https://github.com/fylein/fyle-mobile-app/pull/2503/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L589-R589), [link](https://github.com/fylein/fyle-mobile-app/pull/2503/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L618-R618), [link](https://github.com/fylein/fyle-mobile-app/pull/2503/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L648-R648), [link](https://github.com/fylein/fyle-mobile-app/pull/2503/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L678-R678), [link](https://github.com/fylein/fyle-mobile-app/pull/2503/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L685-R685), [link](https://github.com/fylein/fyle-mobile-app/pull/2503/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L715-R733), [link](https://github.com/fylein/fyle-mobile-app/pull/2503/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L728-R746), [link](https://github.com/fylein/fyle-mobile-app/pull/2503/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L758-R776), [link](https://github.com/fylein/fyle-mobile-app/pull/2503/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L787-R805), [link](https://github.com/fylein/fyle-mobile-app/pull/2503/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L793-R811), [link](https://github.com/fylein/fyle-mobile-app/pull/2503/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L818-R836), [link](https://github.com/fylein/fyle-mobile-app/pull/2503/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L844-R862), [link](https://github.com/fylein/fyle-mobile-app/pull/2503/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L850-R868), [link](https://github.com/fylein/fyle-mobile-app/pull/2503/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L874-R892), [link](https://github.com/fylein/fyle-mobile-app/pull/2503/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L880-R898), [link](https://github.com/fylein/fyle-mobile-app/pull/2503/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L975-R993), [link](https://github.com/fylein/fyle-mobile-app/pull/2503/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L1004-R1022), [link](https://github.com/fylein/fyle-mobile-app/pull/2503/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L1028-R1100), [link](https://github.com/fylein/fyle-mobile-app/pull/2503/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L1050-R1122), [link](https://github.com/fylein/fyle-mobile-app/pull/2503/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L1061-R1133), [link](https://github.com/fylein/fyle-mobile-app/pull/2503/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L1081-R1153), [link](https://github.com/fylein/fyle-mobile-app/pull/2503/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L1092-R1164), [link](https://github.com/fylein/fyle-mobile-app/pull/2503/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L1112-R1184), [link](https://github.com/fylein/fyle-mobile-app/pull/2503/files?diff=unified&w=0#diff-1b36a2db6c6b83c6489707b9032ec98c82a71edc5d235175f2bd80a158ba71d6L1161-R1233))

## Clickup
app.clickup.com

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/fyle/add-edit-expense/add-edit-expense-1.spec.ts [hunk 1]
```diff
@@ -675,20 +675,38 @@ export function TestCases1(getTestBed) {
...
           'information',
-          ['msb-info'],
+          ['msb-info']
         );
         expect(trackingService.showToastMessage).toHaveBeenCalledOnceWith({
           ToastContent: 'Successfully removed the card details from the expense.',
         });
       }));
 
+      it('should go back to my expenses page if etxn is undefined', fakeAsync(() => {
```
Comment (suyashpatil78) [inline src/app/fyle/add-edit-expense/add-edit-expense-1.spec.ts:692]: changes here

File: src/app/fyle/add-edit-expense/add-edit-expense-1.spec.ts [hunk 2]
```diff
@@ -1001,7 +1019,61 @@ export function TestCases1(getTestBed) {
...
+        actionSheetOptions[3].handler();
+        expect(component.markPersonalHandler).toHaveBeenCalledTimes(1);
+        actionSheetOptions[4].handler();
+        expect(component.markDismissHandler).toHaveBeenCalledTimes(1);
+        actionSheetOptions[5].handler();
+        expect(component.removeCCCHandler).toHaveBeenCalledTimes(1);
+        done();
+      });
+
+      it('should get all action sheet options and call titleCasePipe transform method if project_id is defined', (done) => {
```
Comment (suyashpatil78) [inline src/app/fyle/add-edit-expense/add-edit-expense-1.spec.ts:1041]: change here

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 3]
```diff
@@ -586,15 +586,15 @@ export class AddEditExpensePage implements OnInit {
     combineLatest(this.fg.controls.currencyObj.valueChanges, this.fg.controls.tax_group.valueChanges).subscribe(() => {
       if (
         this.fg.controls.tax_group.value &&
-        isNumber(taxGroupControl.value?.percentage) &&
```
Comment (suyashpatil78) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:589]: If we have checked for `this.fg.controls.tax_group.value`, then no need to check for nullity again

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 4]
```diff
@@ -586,15 +586,15 @@ export class AddEditExpensePage implements OnInit {
...
+        isNumber(taxGroupControl.value.percentage) &&
         this.fg.controls.currencyObj.value
       ) {
         const amount =
-          currencyObjControl.value?.amount - currencyObjControl.value?.amount / (taxGroupControl.value?.percentage + 1);
+          currencyObjControl.value.amount - currencyObjControl.value.amount / (taxGroupControl.value.percentage + 1);
```
Comment (suyashpatil78) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:593]: If we have checked for ` this.fg.controls.currencyObj.value`, then no need to check again

File: src/app/fyle/add-edit-expense/add-edit-expense.page.ts [hunk 5]
```diff
@@ -586,15 +586,15 @@ export class AddEditExpensePage implements OnInit {
...
 
         const formattedAmount = this.currencyService.getAmountWithCurrencyFraction(
           amount,
-          currencyObjControl.value?.currency
+          currencyObjControl.value.currency
```
Comment (suyashpatil78) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.ts:597]: same as above

**Review by OmkarJ13 — APPROVED** [review]

File: src/app/fyle/add-edit-expense/add-edit-expense-1.spec.ts [hunk 6]
```diff
@@ -1001,7 +1019,61 @@ export function TestCases1(getTestBed) {
...
+        spyOn(component, 'splitExpCostCenterHandler');
+        spyOn(component, 'markPersonalHandler');
+        spyOn(component, 'markDismissHandler');
+        spyOn(component, 'removeCCCHandler');
+        launchDarklyService.getVariation.and.returnValue(of(true));
+        fixture.detectChanges();
+
+        let actionSheetOptions;
+
+        component.getActionSheetOptions().subscribe((res) => {
```
Comment (hlkavya0213) [inline src/app/fyle/add-edit-expense/add-edit-expense-1.spec.ts:1068]: let's name the variables with proper names based on context. res can be what is the response we get eg: like bankAccountResponse, transactionsResponse etc

Comment (hlkavya0213) [reply src/app/fyle/add-edit-expense/add-edit-expense-1.spec.ts:1068]: here it can be actionSheetOptionsResponse

File: src/app/fyle/add-edit-expense/add-edit-expense-1.spec.ts [hunk 7]
```diff
@@ -1001,7 +1019,61 @@ export function TestCases1(getTestBed) {
...
+          actionSheetOptions = res;
```
Comment (hlkavya0213) [inline src/app/fyle/add-edit-expense/add-edit-expense-1.spec.ts:1069]: why are we assigning res to actionSheetOptions? & I don't see it's usage

File: src/app/fyle/add-edit-expense/add-edit-expense-3.spec.ts [hunk 8]
```diff
@@ -1232,6 +1232,34 @@ export function TestCases3(getTestBed) {
...
+        component.orgUserSettings$ = of(orgUserSettingsData);
+        categoriesService.getAll.and.returnValue(of(orgCategoryData1));
+        component.homeCurrency$ = of('USD');
+        dateService.getUTCDate.and.returnValue(new Date('2023-01-24T11:30:00.000Z'));
+        spyOn(component, 'getInstaFyleImageData').and.returnValue(of(instaFyleData1));
+        recentLocalStorageItemsService.get.and.resolveTo(undefined);
+        component.recentlyUsedValues$ = of(recentlyUsedRes);
+        fixture.detectChanges();
+
+        component.getNewExpenseObservable().subscribe((res) => {
```
Comment (hlkavya0213) [inline src/app/fyle/add-edit-expense/add-edit-expense-3.spec.ts:1247]: same as above

File: src/app/fyle/add-edit-expense/add-edit-expense-1.spec.ts (hunk 6 above)
Comment (suyashpatil78) [reply src/app/fyle/add-edit-expense/add-edit-expense-1.spec.ts:1068]: yes, changing...

**Review by hlkavya0213 — APPROVED** [review]
